### PR TITLE
Bump for last node 8

### PR DESCRIPTION
**Pull request by shiftkey**
## Overview

Appveyor has the latest version of Node 8 installed on its agents, so we currently waste time on CI builds uninstalling that to choose the previous version specified in config, just for running scripts:

```
Install-Product node $env:nodejs_version $env:platform
Uninstalling node 8.12.0 (x86)...
Installing node 8.11.4 (x64)...
```

This is one of our bottlenecks for getting feedback on builds, so hopefully this saves us some time.

## Description

This touches all the places where we mention `8.11.x`, so it's a good reference for what we need to update when we want to move to Node 10 in #5876 

TODO:

 - [x] Azure Pipelines Windows is reverting to Python 3 (Python2 is available, just not on PATH)
 - [x] fix comment in docs

## Release notes

Notes: no-notes

### CODE REVIEW

**Comment by outofambit** [comment]
oops sorry, i got excited!

File: docs/contributing/setup-macos.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ $ node -v
 
 If you see an error about being unable to find `node`, that probably means you don't have any Node tools installed. You can install Node LTS (the version we need) from the [Node.js website](https://nodejs.org/en/download/) and restart your shell.
 
-If you see the output `v8.11.x` (where `x` is any number), you're good to go.
+If you see the output `v8.12.0` (where `x` is any number), you're good to go.
```
Comment (j-f1) [inline docs/contributing/setup-macos.md]: Should “(where `x` is any number)” be removed?

**Comment by outofambit** [comment]
exciting!! i think we also need to change https://github.com/desktop/desktop/blob/952a9b15f0af7f331998a7e415f165a39ca77c0b/package.json#L52

**Comment by shiftkey** [comment]
@outofambit I've got mixed feelings on that, but mostly err on the side of "do we need to force everyone to update?" This original issue came from trying to improve CI scenarios, and I'm fine with not forcing contributors to upgrade too (this is just for tooling).

**Comment by outofambit** [comment]
@shiftkey 👍 i don't feel strongly about this. in the future though i think we should leave off the y version in `package.json` if we don't want to update it along with everything else

**Comment by shiftkey** [comment]
> i think we should leave off the y version in package.json if we don't want to update it along with everything else

@outofambit yep, I'll keep this in mind whenever we get around to #5876, but I had to add this in because of a specific issue with a package: https://github.com/desktop/desktop/pull/4943#discussion_r195795606

**Pull request merged**